PhysicsTools/PatUtils/python/tools/eGammaCorrection.py

```diff
@@ -0,0 +1,111 @@
+import FWCore.ParameterSet.Config as cms
+
+from FWCore.GuiBrowsers.ConfigToolBase import *
+#import PhysicsTools.PatAlgos.tools.helpers as configtools
+
+
+def eGammaCorrection(process,
+                     electronCollection,
+                     photonCollection,
+                     corElectronCollection,
+                     corPhotonCollection,
+                     metCollections,
+                     pfCandMatching=False,
+                     pfCandidateCollection="",
+                     #correctAlreadyExistingMET,
+                     corMetName="corEGSlimmedMET",
+                     postfix=""
+                     ):
+
+    process.load("PhysicsTools.PatAlgos.cleaningLayer1.photonCleaner_cfi")
+    #cleaning the bad collections
+    cleanedPhotonCollection="cleanedPhotons"+postfix
+    cleanPhotonProducer = getattr(process, "cleanPatPhotons").clone( 
+                    src = photonCollection,
+                    
+                    )
+    cleanPhotonProducer.checkOverlaps.electrons.src = electronCollection
+    cleanPhotonProducer.checkOverlaps.electrons.requireNoOverlaps=cms.bool(True)
+    
+    #cleaning the good collections
+    cleanedCorPhotonCollection="cleanedCorPhotons"+postfix
+    cleanCorPhotonProducer = getattr(process, "cleanPatPhotons").clone( 
+                    src = corPhotonCollection
+                    )
+    cleanCorPhotonProducer.checkOverlaps.electrons.src = corElectronCollection
+    cleanCorPhotonProducer.checkOverlaps.electrons.requireNoOverlaps=cms.bool(True)
+    #cleanCorPhotonProducer = cms.EDProducer("CandPtrProjector", 
+    #src = cms.InputTag(corPhotonCollection),
+    #veto = cms.InputTag(corElectronCollection)
+    #)
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant